cassandra/src/main/java/org/hibernate/ogm/datastore/cassandra/CassandraDialect.java

```diff
@@ -210,12 +236,61 @@ public void removeTuple(EntityKey key, TupleContext tupleContext) {
 
 	@Override
 	public Association getAssociation(AssociationKey key, AssociationContext associationContext) {
-		return null; // TODO
+
+		String table = key.getTable();
+		Table tableMetadata = provider.getMetaDataCache().get( key.getTable() );
+		List<Column> tablePKCols = (List<Column>) tableMetadata.getPrimaryKey().getColumns();
+
+		Select select = select().all().from( quote( key.getTable() ) );
+		Select.Where selectWhere = select.where( eq( quote( key.getColumnNames()[0] ), QueryBuilder.bindMarker() ) );
+		for ( int i = 1; i < key.getColumnNames().length; i++ ) {
+			selectWhere = selectWhere.and( eq( quote( key.getColumnNames()[i] ), QueryBuilder.bindMarker() ) );
+		}
+
+		boolean requiredFiltering = false;
```

Comment:
cassandra is designed to deal with Big Data. The general rule of thumb is that query execution costs should be proportional to result set size. It's an anti-pattern to scan a potentially large amount of data and discard most of it, since the disk I/O costs at scale are prohibitive. To force the engine to do that you need to add an explicit override 'ALLOW FILTERING' to your query.  Actually I could just tack that on to all the queries, but I like knowing which ones we eventually want to rework so they don't need it at all by e.g. denormalizing the storage into alternatively indexed tables.